.github/workflows/CI.yml

```
@@ -1,13 +1,14 @@
 name: Rust CI
 
-on: [push, pull_request]
+on:[push, pull_request]
```

Is there a specific reason why you want to keep the push trigger here? Note that it places an unnecessary stress on pipeline resources, especially when all branches are eligible.

```suggestion
on: [pull_request]
```

---

From the other PR i am seeing that the CI is done twice and i think it is because of that last CI 
It is now clear to me